### PR TITLE
use a class for the chia cli context data

### PR DESCRIPTION
**Pull request by altendky**
### Draft For:
- [x] https://github.com/Chia-Network/chia-blockchain/pull/18833
- [x] https://github.com/Chia-Network/chia-blockchain/pull/18641
- [x] https://github.com/Chia-Network/chia-blockchain/pull/19047

### CODE REVIEW

File: chia/cmds/cmd_classes.py [hunk 1]
```diff
@@ -48,6 +62,29 @@ def option(*param_decls: str, **kwargs: Any) -> Any:
...
+@dataclasses.dataclass
+class ChiaCliContext:
+    context_dict_key: ClassVar[str] = "_chia_cli_context"
+
+    root_path: pathlib.Path = DEFAULT_ROOT_PATH
+    expected_prefix: Optional[str] = None
+    rpc_port: Optional[int] = None
+    keys_fingerprint: Optional[int] = None
+    keys_filename: Optional[str] = None
+    expected_currency_prefix: Optional[str] = None
```
Comment (Quexington) [inline chia/cmds/cmd_classes.py:75]: Nittest of nits but can we change this to address prefix?  Not everything being described will be a "currency" (think NFTs, DIDs, etc.)

Also, not sure (yet, maybe) what the difference is between this and expected_prefix.

File: chia/cmds/sim.py [hunk 2]
```diff
@@ -34,10 +35,9 @@
 )
 @click.pass_context
 def sim_cmd(ctx: click.Context, rpc_port: Optional[int], root_path: str, simulator_name: str) -> None:
-    ctx.ensure_object(dict)
-    ctx.obj["root_path"] = Path(root_path) / simulator_name
-    ctx.obj["sim_name"] = simulator_name
```
Comment (Quexington) [inline chia/cmds/sim.py:39]: This is an intentional deletion that does not get re-added in the class context?

File: chia/cmds/cmd_classes.py (hunk 1 above)
Comment (altendky) [reply chia/cmds/cmd_classes.py:75]:
```suggestion
    expected_address_prefix: Optional[str] = None
```

File: chia/cmds/sim.py (hunk 2 above)
Comment (altendky) [reply chia/cmds/sim.py:39]: the `simulator_name` value from the cli is used for creating a sub-'root path'.  it's assignment into the context dict was unused and thus removed.

**Review by Quexington — APPROVED** [review]

**Review by emlowe — APPROVED** [review]

**Pull request merged**